packages/i18n-utils/src/date-utils.ts

```diff
@@ -125,3 +125,37 @@ export const getNumericDateTimeString = ( timestamp: number, locale = 'en' ) =>
 		return getISODateString( timestamp );
 	}
 };
+
+/**
+ * Get localized first day of week 1–7, 1 being Monday and 7 being Sunday.
+ * Uses vanilla API for retrieving value from `Locale.prototype.getWeekInfo()`,
+ * or the currently more supported `Locale.prototype.weekInfo`.
+ * As of 05/2025, Firefox does not support either.
+ *
+ * If you need full support across browsers, you could load a polyfill first:
+ * https://github.com/bart-krakowski/get-week-info-polyfill
+ * @param {string} locale Locale slug
+ * @param {number} defaultFirstDayOfWeek Default first day of week if browser API is not available or locale is incorrect.
+ * @returns {number} First day of week 1–7, 1 being Monday and 7 being Sunday.
+ */
+export const getNumericFirstDayOfWeek = ( locale: string, defaultFirstDayOfWeek: number = 1 ) => {
```

Comment:
@Automattic/i18n I wanted to check with you that adding a simple utility like this is fine. The purpose isn't so significant that it doesn't justify a heavy polyfill, but it's a nice UX touch that helps to localize UI.

Reply:
The approach looks good to me. It's unfortunate that there is no support for Firefox yet, but at least all of the other major browsers are supported.